### PR TITLE
fix `split list` when separater is the first element of list

### PR DESCRIPTION
**Pull request by WindSoilder**
# Description

Fixes: #7278

# User-Facing Changes

_(List of all changes that impact the user experience here. This helps us keep track of breaking changes.)_

# Tests + Formatting

Don't forget to add tests that cover your changes.

Make sure you've run and fixed any issues with these commands:

- `cargo fmt --all -- --check` to check standard code formatting (`cargo fmt --all` applies these changes)
- `cargo clippy --workspace -- -D warnings -D clippy::unwrap_used -A clippy::needless_collect` to check that you're using the standard code style
- `cargo test --workspace` to check that all tests pass

# After Submitting

If your PR had any user-facing changes, update [the documentation](https://github.com/nushell/nushell.github.io) after the PR is merged, if necessary. This will help us keep the docs up to date.

### CODE REVIEW

File: crates/nu-command/src/strings/split/list.rs [hunk 1]
```diff
@@ -95,6 +95,24 @@ impl Command for SubCommand {
                     span: Span::test_data(),
                 }),
             },
+            Example {
+                description: "Split a list of chars into two lists",
+                example: "[a, b, c, d, e, f, g] | split list a",
```
Comment (fdncred) [inline crates/nu-command/src/strings/split/list.rs:100]: Would you mind changing this to doing some splitting other than just removing the first item in the list? Maybe something like `[a, b, c, d, a, e, f, g] | split list a`? At least then we end up with two lists, which I think is the intent. The way this example is now could be confusing.

**Comment by fdncred** [comment]
Other than my one comment, I tested this and it fixes the bug and works as advertised.

File: crates/nu-command/src/strings/split/list.rs (hunk 1 above)
Comment (WindSoilder) [reply crates/nu-command/src/strings/split/list.rs]: Ok, that's a good point :)

**Review by fdncred — APPROVED** [review]
LGTM! Thanks.

**Pull request merged**